backup.sh

```diff
@@ -136,17 +136,21 @@ parse-file-timestamp () {
 
 # Delete a backup
 delete-backup () {
-  local BACKUP=$1
-  rm $BACKUP_DIRECTORY/$BACKUP
+  local BACKUP=$2
+  local BACKUP_DIRECTORY=$1
+  rm "$BACKUP_DIRECTORY/$BACKUP"
   message-players "Deleted old backup" "$BACKUP"
 }
 
 # Sequential delete method
 delete-sequentially () {
-  local BACKUPS=($(ls $BACKUP_DIRECTORY))
+  local BACKUP_DIRECTORY=$1
+  local WORLD_NAME=$2
+  local BACKUPS=($(ls $BACKUP_DIRECTORY | grep "$WORLD_NAME\.tar$COMPRESSION_FILE_EXTENSION\$"))
```

Comment:
I'm still skeptical of this use of `grep`. Consider if someone has worlds named "nether_world" and "world". Then your files may look like
- "2020-06-14_00-00-00_nether_world.tar.gz"
- "2020-06-14_00-00-00_world.tar.gz"
and both would be listed when grep occurs for "world", leading to unintended deletion. 

I think this could be fixed with a stricter regex pattern in grep. For maintainability, listing backups by world should probably be put into its own function, so that you don't have to repeat code (I see a similar grep is used on lines 153 and 202).

Reply:
one would _hope_ that they would use separate backup directories per world? 
```
BACKUP_DIRECTORY[0]="/media/server/ExternalStorage/Backups/PrivateSurvivalBackups"
```

Reply:
If they're in different backup directories then it would (probably) be fine, but as it is this PR allows you to have the option to just specify one and it will use that one for all of them (see line 264). As a fix, I think a more defensive regex would do (something that incorporates matching the expected date format).